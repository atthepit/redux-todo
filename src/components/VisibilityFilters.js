import React from 'react'

const VisibilityFilters = ({ onClick }) =>
  <div>
    <button onClick={ () => onClick(`SHOW_ALL`) } >{`Show All`}</button>
    <button onClick={ () => onClick(`SHOW_ACTIVE`) } >{`Active`}</button>
    <button onClick={ () => onClick(`SHOW_COMPLETED`) } >{`Completed`}</button>
  </div>

VisibilityFilters.propTypes = {
  onClick: React.PropTypes.func,
}

export default VisibilityFilters
