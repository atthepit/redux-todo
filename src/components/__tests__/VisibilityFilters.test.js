import React from 'react'
import VisibilityFilters from '../VisibilityFilters'
import renderer from 'react-test-renderer'

describe(`VisibilityFilters component`, () => {
  it(`renders correctly`, () =>{
    const props = {
      onClick: jest.fn(),
    }

    const tree = renderer.create(
      <VisibilityFilters { ...props } />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
