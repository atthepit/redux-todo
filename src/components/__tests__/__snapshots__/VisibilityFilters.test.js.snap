exports[`VisibilityFilters component renders correctly 1`] = `
<div>
  <button
    onClick={[Function]}>
    Show All
  </button>
  <button
    onClick={[Function]}>
    Active
  </button>
  <button
    onClick={[Function]}>
    Completed
  </button>
</div>
`;
